storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -522,6 +522,18 @@ public TimeIndex timeIndex() {
 
         // Visible for testing
         public TransactionIndex txnIndex() {
+            if (txnIndex.isClosed()) {
```

Comment:
Nit: Maybe it makes sense to add a debug log here to indicate that the log was closed and is being reopened? Might be useful to debug any performance issues.